detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,243 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eks
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/semconv"
+)
+
+const (
+	k8sSvcURL          = "https://kubernetes.default.svc"
+	k8sTokenPath       = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	k8sCertPath        = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	authConfigmapPath  = "/api/v1/namespaces/kube-system/configmaps/aws-auth"
+	cwConfigmapPath    = "/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info"
+	defaultCgroupPath  = "/proc/self/cgroup"
+	containerIDLength  = 64
+	millisecondTimeOut = 2000
```

Comment:
Perhaps this should be `timeoutMillis`?  When I see `millisecondTimeOut` used in the code I think it is set for 1ms.